File: src/main/java/com/blackcowmoo/moomark/auth/model/oauth2/GoogleTokenResponse.java
package com.blackcowmoo.moomark.auth.model.oauth2;

import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

@ToString
@NoArgsConstructor
@Getter
public class GoogleTokenResponse {
  @JsonProperty("access_token")
  private String accessToken;
  @JsonProperty("expires_in")
  private Integer expiresIn;
  private String scope;
  @JsonProperty("token_type")
  private String tokenType;
  @JsonProperty("id_token")
  private String idToken;

  public Boolean isExpired() {
    return this.expiresIn <= 0;
  }
}

File: src/main/kotlin/com/blackcowmoo/moomark/auth/model/oauth2/GoogleTokenResponse.kt
package com.blackcowmoo.moomark.auth.model.oauth2

import com.fasterxml.jackson.annotation.JsonProperty

data class GoogleTokenResponse(
    @JsonProperty("access_token") val accessToken: String? = null,
    @JsonProperty("expires_in") val expiresIn: Int? = null,
    val scope: String? = null,
    @JsonProperty("token_type") val tokenType: String? = null,
    @JsonProperty("id_token") val idToken: String? = null
) {
    fun isExpired(): Boolean {
        return expiresIn?.let { it <= 0 } ?: true
    }
}